js/modules/k6/execution/execution_test.go

```diff
@@ -142,21 +142,24 @@ func TestVUTags(t *testing.T) {
 			assert.Equal(t, "vu101", val.String())
 		})
 
-		t.Run("DiscardWrongTypeRaisingError", func(t *testing.T) {
+		t.Run("DiscardWrongTypeAndRaisingError", func(t *testing.T) {
 			t.Parallel()
 
 			tenv := setupTagsExecEnv(t)
 			state := tenv.Module.vu.State()
 			state.Options.Throw = null.BoolFrom(true)
 			require.NotNil(t, state)
 
-			// array
-			_, err := tenv.Runtime.RunString(`exec.vu.tags["custom-tag"] = [1, 3, 5]`)
-			require.Contains(t, err.Error(), "only String, Boolean and Number")
+			cases := []string{
+				`[1, 3, 5]`,             // array
+				`{f1: "value1", f2: 4}`, // object
+			}
 
-			// object
-			_, err = tenv.Runtime.RunString(`exec.vu.tags["custom-tag"] = {f1: "value1", f2: 4}`)
-			require.Contains(t, err.Error(), "only String, Boolean and Number")
+			for _, val := range cases {
+				_, err := tenv.Runtime.RunString(`exec.vu.tags["custom-tag"] = ` + val)
+				require.Contains(t, err.Error(), "TypeError:")
```

Comment:
```suggestion
				_, err := tenv.Runtime.RunString(`exec.vu.tags["custom-tag"] = ` + val)
				require.Error(t, err)
				require.Contains(t, err.Error(), "TypeError:")
```

otherwise if this doesn't produce an error it will panic the tests